modules/contact/test/tests/simple_contact/two_block_compress/two_equal_blocks_compress_2d.i

```diff
@@ -0,0 +1,278 @@
+  []
+[]
+
+[Modules/TensorMechanics/Master]
+  [all]
+    strain = FINITE
+    incremental = true
+    add_variables = true
+    block = '1 2'
+  []
+[]
+
+[Functions]
+  [horizontal_movement]
+    type = PiecewiseLinear
+    x = '0 1.0'
+    y = '0 0.4'
+  []
+  [vertical_movement]
+    type = PiecewiseLinear
+    x = '0 1.0'
+    y = '0 0'
+  []
+[]
+
+[BCs]
+  [push_left_x]
+    type = FunctionDirichletBC
+    variable = disp_x
+    boundary = 13
+    function = horizontal_movement
+  []
+  [fix_right_x]
+    type = DirichletBC
+    variable = disp_x
+    boundary = 21
+    value = 0.0
+  []
+  [fix_right_y]
+    type = DirichletBC
+    variable = disp_y
+    boundary = 21
+    value = 0.0
+  []
+  [push_left_y]
+    type = FunctionDirichletBC
+    variable = disp_y
+    boundary = 13
+    function = vertical_movement
+  []
+[]
+
+[Materials]
+  [elasticity_tensor_left]
+    type = ComputeIsotropicElasticityTensor
+    block = 1
+    youngs_modulus = 1.0e6
+    poissons_ratio = 0.3
+  []
+  [stress_left]
+    type = ComputeFiniteStrainElasticStress
+    block = 1
+  []
+
+  [elasticity_tensor_right]
+    type = ComputeIsotropicElasticityTensor
+    block = 2
+    youngs_modulus = 1.0e6
+    poissons_ratio = 0.3
+  []
+  [stress_right]
+    type = ComputeFiniteStrainElasticStress
+    block = 2
+  []
+[]
+
+[Constraints]
+  [normal_lm]
+    type = ComputeWeightedGapLMMechanicalContact
+    primary_boundary = '23'
+    secondary_boundary = '11'
+    primary_subdomain = 'primary_lower'
+    secondary_subdomain = 'secondary_lower'
+    variable = normal_lm
+    disp_x = disp_x
+    disp_y = disp_y
+    use_displaced_mesh = true
+    correct_edge_dropping = true
+  []
+  [normal_x]
+    type = NormalMortarMechanicalContact
+    primary_boundary = '23'
+    secondary_boundary = '11'
+    primary_subdomain = 'primary_lower'
+    secondary_subdomain = 'secondary_lower'
+    variable = normal_lm
+    secondary_variable = disp_x
+    component = x
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+    correct_edge_dropping = true
+  []
+  [normal_y]
+    type = NormalMortarMechanicalContact
+    primary_boundary = '23'
+    secondary_boundary = '11'
+    primary_subdomain = 'primary_lower'
+    secondary_subdomain = 'secondary_lower'
+    variable = normal_lm
+    secondary_variable = disp_y
+    component = y
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+    correct_edge_dropping = true
+  []
+[]
+
+[Preconditioning]
+  [SMP]
+    type = SMP
+    full = true
+  []
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = 'PJFNK'
+
+  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package -pc_factor_shift_type '
+                        '-pc_factor_shift_amount'
+  petsc_options_value = 'lu    superlu_dist nonzero 1e-10'
+
+  line_search = 'none'
+
+  dt = 0.1
+  dtmin = 0.01
+  end_time = 1.0
+
+  l_max_its = 20
+
+  nl_max_its = 20
+  nl_rel_tol = 1e-6
+  nl_abs_tol = 1e-8
+  snesmf_reuse_base = false
```

Comment:
For future reference, @lindsayad and I came to the realization that this setting shouldn't be needed for mortar mechanical contact when the recommended `weighted` objects are used.